Disher/Models/Recipe.m

```diff
@@ -27,4 +27,33 @@ + (Recipe *) initWithRecipe:(NSString *) name withURL:(NSString *) imgURL withSo
     return newRecipe;
 }
 
++ (void) getRecipeInfo:(NSString *)recipeID withSource:(NSString *) source withCompletion:(void(^)(NSDictionary *recipeInformation))completionHandler {
+    NSURL *url;
+    if ([source isEqualToString:@"Spoonacular"]) {
+        NSString *path = [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"];
+        NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: path];
+        NSString *key = [dict objectForKey: @"spoon_key"];
+        NSString *apiKeyArg = [NSString stringWithFormat:@"?apiKey=%@", key];
+        url = [NSURL URLWithString:[NSString stringWithFormat:@"https://api.spoonacular.com/recipes/%@/information%@", recipeID, apiKeyArg]];
+        NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
+        NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
+        NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+               NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
+               completionHandler(dataDictionary);
+            
+        }];
+        [task resume];
+    }
+    else if ([source isEqualToString:@"TheMealDB"]) {
+        url = [NSURL URLWithString:[NSString stringWithFormat:@"https://www.themealdb.com/api/json/v1/1/lookup.php?i=%@", recipeID]];
```

Comment:
for example, this API URL; if it ever changes, it's easier to edit it from one single spot.